docs/high_availability_installation_guide.md

```diff
@@ -0,0 +1,284 @@
+ - Harbor database(MariaDB)
+ - Clair database(PostgresSQL)
+ - Notary database(MariaDB)
+ - Redis
+Making those services highly available can depend the high available implementation for each application. 
+
+## High Availability Architecture 
+
+Again, this architecture is for advice only. 
+![HA](img/ha/Architecture.png)
+As seen in the figure, components involved in the architecture are: 
+ 
+**VIP**: Virtual IP. The Harbor user will access Harbor through this virtual IP address. This VIP will only active on one load balancer node at the same time. It will automatically switch to the other node if the active loadbalancer node is down. 
+
+**LoadBalancer01 and 02**: They together compose as a group which avoid single point failure of load balancer nodes. Keepalived is installed on both load balancer nodes. The two Keepalived instance will form a VRRP group to provide the VIP and ensure the VIP only shows on one node at the same time. The LVS component in Keepalived is responsible for balance the requests between different Harbor servers according to the routing algorithm.  
+
+**Harbor server 1..n**: These are the running Harbor instances. They are in active-active mode. User can setup multiple nodes according to their workload. 
+
+**Harbor DB cluster**: The MariaDB is used by Harbor to store user authentication information, image metadata information and so on. User should follow its best practice to make it HA protected. 
+
+**Clair DB cluster**: The PostgreSQL is used by Clair to store the vulnerability data which will be used by scanning the images. User should follow it's best practice to make it HA protected 
+
+**Shared Storage**: The shared storage is used for storing Docker Volumes used by Harbor. Images pushed by users are actually stored in this shared storage. The shared storage makes sure that multiple Harbor instances have consistent storage backend. Shared Storages can be Swift, NFS, S3, azure, GCS Ceph or OSS. User should follow its best practice to make it HA protect. 
+
+**Redis**: The purpose of having Redis is to store UI session data and store the registry metadata cache. When one Harbor instance fails or the load balancer routes a user request to another Harbor instance, any Harbor instance can query the Redis to retrieve session information to make sure the end-user has a continued session. User should follow the best practice of Redis to make it HA protect.  
+
+From the above high availability architecture, we can see that we don't setup LB per stateless services. Instead we group those stateless service as a group. The communicate between each services are protected by host based docker network with isolation. **Please Note!** As the component communicate with each other through rest API. You can always define the group granularity according to your use scenarios. 
+
+### Limitation 
+
+Currently Harbor doesn’t support and Notary in HA scenario. Which means it don't support Content trust for this HA setup. 
```

Comment:
support and Notary => support Notary ?
. Which means it don't support Content trust for this HA setup.  => which means Content Trust is not supported in Harbor HA scenario.